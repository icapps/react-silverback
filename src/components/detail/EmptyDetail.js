import React from 'react';
import PropTypes from 'prop-types';
import { strings } from '../../utils';
import './emptyDetail.scss';

const arrowLeft = require('../../assets/images/arrow-left.svg');

const EmptyDetail = props => {
  const overview = window.location.pathname.split('/')[1];
  return (
    <main className="detail-empty col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3">
      <div className="container">
        <div className="back text-primary" onClick={props.history.goBack}>
          <img src={arrowLeft} alt="back" />
          <span>{`${overview.charAt(0).toUpperCase()}${overview.slice(1)}`}</span>
        </div>
        <h3 className="title">{strings.GENERAL_ERROR}</h3>
      </div>
    </main>
  );
};

EmptyDetail.propTypes = {
  history: PropTypes.object.isRequired,
};

export default EmptyDetail;
