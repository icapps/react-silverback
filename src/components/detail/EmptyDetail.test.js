import React from 'react';
import { shallow, configure } from 'enzyme';
import EmptyDetail from './EmptyDetail';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('EmptyDetail Component', () => {
  it('should render a EmptyDetail component', () => {
    const wrapper = shallow(
      <EmptyDetail
        history={{ goBack: () => { } }}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
