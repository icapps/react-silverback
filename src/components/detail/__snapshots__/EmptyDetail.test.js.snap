// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyDetail Component should render a EmptyDetail component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyDetail
    history={
        Object {
            "goBack": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <div
                className="back text-primary"
                onClick={[Function]}
        >
                <img
                        alt="back"
                        src="arrow-left.svg"
                />
                <span />
        </div>
        <h3
                className="title"
        >
                Something went wrong!
        </h3>
</div>,
      "className": "detail-empty col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back text-primary"
            onClick={[Function]}
>
            <img
                        alt="back"
                        src="arrow-left.svg"
            />
            <span />
</div>,
          <h3
            className="title"
>
            Something went wrong!
</h3>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="back"
                src="arrow-left.svg"
/>,
              <span />,
            ],
            "className": "back text-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "back",
                "src": "arrow-left.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something went wrong!",
            "className": "title",
          },
          "ref": null,
          "rendered": "Something went wrong!",
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <div
                    className="back text-primary"
                    onClick={[Function]}
          >
                    <img
                              alt="back"
                              src="arrow-left.svg"
                    />
                    <span />
          </div>
          <h3
                    className="title"
          >
                    Something went wrong!
          </h3>
</div>,
        "className": "detail-empty col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="back text-primary"
              onClick={[Function]}
>
              <img
                            alt="back"
                            src="arrow-left.svg"
              />
              <span />
</div>,
            <h3
              className="title"
>
              Something went wrong!
</h3>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="back"
                  src="arrow-left.svg"
/>,
                <span />,
              ],
              "className": "back text-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "back",
                  "src": "arrow-left.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Something went wrong!",
              "className": "title",
            },
            "ref": null,
            "rendered": "Something went wrong!",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
